tools/build_config/build.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+
+set -o errexit  # exit early on any error
+set -o nounset  # raise error using unused variables
+
+FALSE=0
+TRUE=1
+CMAKE_GENERATOR="Unix Makefiles"
+HORACE_ROOT=$(realpath $(dirname "$0")/../..)
+MATLAB_ROOT="$(realpath $(dirname $(readlink -f $(which matlab)))/..)"
+
+# set default parameter values
+build=$FALSE
+test=$FALSE
+package=$FALSE
+build_tests="ON"
+build_config='Release'
+build_dir="$HORACE_ROOT/build"
+install_dir="$HORACE_ROOT/install"
+
+
+function echo_and_run {
+    echo "+ $1"
+    eval "$1"
+}
+
+function warning {
+    echo -e "\e[33m$1\e[0m"
+}
+
+function run_build() {
+    echo_and_run "cd ${build_dir}"
```

Comment:
it would be cleaner to pass the variables in as arguments (`$build_config`; `$build_dir`) rather than use global variables

```
$build_dir = $1
$build_config = $2
```
and similarly `$install_dir` in the package step.
You can use `local` to control the scope of these function-scoped variables